flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/ExecNodeContext.java

```diff
@@ -105,6 +105,9 @@ private ExecNodeContext(@Nullable Integer id, String name, Integer version) {
     public ExecNodeContext(String value) {
         this.id = null;
         String[] split = value.split("_");
+        if ("null".equals(split[0]) || "null".equals(split[1])) {
```

Comment:
I am not very familiar with this code; you  are saying that the value can come through as either null_xxx or yyy_null. 
Is there not a way to test for these nulls prior to calling this.

Also I wonder if you should check for the failure of Integer.valueOf(split[1]); and give a nicer message in that case.

Reply:
Please see the discussion here: https://github.com/apache/flink/pull/23488#discussion_r1345505603